applicationconfig/client/src/main/java/com/azure/applicationconfig/models/ConfigurationSetting.java

```diff
@@ -57,18 +59,17 @@ public ConfigurationSetting() {
      * @param other The configuration setting to copy values from.
      */
     public ConfigurationSetting(ConfigurationSetting other) {
-        this.key(other.key)
-            .label(other.label)
-            .etag(other.etag)
-            .value(other.value)
-            .contentType(other.contentType);
+        this.key = other.key();
+        this.label = other.label();
+        this.etag = other.etag();
+        this.value = other.value();
+        this.contentType = other.contentType();
+        this.locked = other.isLocked();
+        this.lastModified = other.lastModified();
 
         if (other.tags() != null) {
-            other.tags(new HashMap<>(other.tags()));
+            this.tags = new HashMap<>(other.tags());
         }
-
-        this.locked = other.isLocked();
-        this.lastModified = other.lastModified();
     }
```

Comment:
Copy constructor that appears to only be used in unit tests, was changing it since there was a minor bug in it and it seemed strange that the constructor was setting the properties via the setter methods.